src/Backups/BackupsWorker.cpp

```diff
@@ -18,37 +18,86 @@
 #include <Common/Exception.h>
 #include <Common/Macros.h>
 #include <Common/logger_useful.h>
-#include <Common/scope_guard_safe.h>
 #include <Common/setThreadName.h>
 
 
 namespace DB
 {
 
-namespace ErrorCodes
-{
-    extern const int LOGICAL_ERROR;
-}
-
 namespace
 {
     /// Coordination status meaning that a host finished its work.
-    constexpr const char * kCompletedCoordinationStatus = "completed";
+    constexpr const char * kCompletedStage = "completed";
+
+    std::shared_ptr<IBackupCoordination> makeBackupCoordination(const String & coordination_zk_path, const ContextPtr & context, bool is_internal_backup)
+    {
+        if (!coordination_zk_path.empty())
+        {
+            auto get_zookeeper = [global_context = context->getGlobalContext()] { return global_context->getZooKeeper(); };
```

Comment:
Ephemeral nodes are used only to check if a remote node is still alive, because we don't want the initiator to wait forever for a node which is not alive anymore.

I thought about using ephemeral nodes for backup coordination anywhere but it appeared that ephemeral nodes cannot have ephemeral children.